memory/memtester.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Basheer K<basheer@linux.vnet.ibm.com>
+#
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import build
+from avocado.utils import memory
+from avocado.utils import git
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Memtester(Test):
+    """
+    1.memtester  is  an  effective  userspace  tester  for stress-testing the
+      memory subsystem.  It is very effective  at  finding  intermittent  and
+      non-deterministic  faults.
+    2.memtester must be run with  root  privileges  to  mlock(3)  its  pages.
+      Testing  memory  without locking the pages in place is mostly pointless
+      and slow.
+    """
+
+    def setUp(self):
+        sm = SoftwareManager()
+        detected_distro = distro.detect()
+        deps = ['gcc', 'make']
+        if detected_distro.name == "SuSE":
+            deps.append('git-core')
+        else:
+            deps.append('git')
+        for pkg in deps:
+            if not sm.check_installed(pkg) and not sm.install(pkg):
+                self.error('%s is needed for the test to be run' % pkg)
+
+        git.get_repo('https://github.com/jnavila/memtester.git',
+                     destination_dir=self.srcdir)
```

Comment:
I'd  suggest using fetch asset and url can be `https://github.com/jnavila/memtester/archive/master.zip `
 It can be cached and downloads only the snapshot, not the whole history... and use `expire` param as well  which help that avocado will download it again if expired.